lib/realtime_web/channels/user_socket.ex

```diff
@@ -25,10 +26,10 @@ defmodule RealtimeWeb.UserSocket do
       log_level =
         params
         |> Map.get("log_level", @default_log_level)
-        |> case do
+        |> then(fn
           "" -> @default_log_level
           level -> level
-        end
+        end)
```

Comment:
Here's a simple comparison of different clauses, which is exactly what `case do` was designed for. Why would you want to wrap pattern matching in a fun?

Reply:
another one that would be a long discussion 😅

IMO this is a bit of an anti pattern as it leads to code to be "piled up" into the case while `then( fn ` will make it more "intuitive" to extract this to functions when the code requires. Plus it follow the idea of piping into functions

again, personal opinion which in this specific scenario I have no strong opinion about since it's a small amount of code.

Reply:
I don't want to start a holy war, I'm just curious because you replaced the existing code. I thought you had a strong opinion on this. From my perspective, doing pattern matching inside a lambda is actually an anti-pattern when Elixir already provides specific constructs for this purpose

Reply:
For what is worth, and not discussing style, the `case do` is guaranteed to always be faster as it doesn't require creating a function and then invoking it. We may be able to optimize the function one, but it is not guaranteed. :)

Reply:
That said, my comment above is explicitly about `then + fn` vs `case`. It is also completely fine to pattern match inside functions. :)

Reply:
Ny opinion is more on the side that pipes handle functions and as such then+fn makes more "sense" with that idea in mind

Plus it makes it easier to guide the person to the discussion of "should I extract it to a private function?" 

No strong opinions either 😁 whatever makes the team happier I'm in but this is my rationale

Reply:
@josevalim thanks for the additional comments. I apologize if that irrelevant discussion distracted you :D

Reply:
Nah, no worries. It is still good language feedback. :)